### PR TITLE
Release note fixes

### PR DESCRIPTION
**Pull request by jseldess**
Based on feedback from @tamird in https://github.com/cockroachdb/docs/pull/728

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/729)

### CODE REVIEW

File: beta-20161006.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ summary: Additions and changes in CockroachDB version beta-20161006.
 
 ## Oct 6, 2016
 
-Although this week's release contains only a few internal changes and doc updates, note that we are preparing to merge numerous features from our `develop` branch to `master` for inclusion in an upcoming release.  
+Although this week's release contains only a few internal changes and doc updates, note that we are preparing to release numerous features currently on our `develop` branch.
```
Comment (tamird) [inline beta-20161006.md]: not to be overly pedantic, but these are not on our develop branch; that branch is already deleted.

i would suggest avoiding being overly specific here.

Comment (jseldess) [inline beta-20161006.md]: Good point. Can you suggest better language than "we are preparing to release numerous features" though? I'd prefer some specificity. What about something like: "we are preparing to release numerous features that were developed but not released during our recent focus on core stability."

Comment (tamird) [inline beta-20161006.md]: "This week's release does not contain many user-visible changes. This is an artifact of our recent stability efforts, which funneled user-visible changes to a secondary development branch. With the conclusion of the concerted stability effort, we are preparing to include these features in next week's release."

Comment (jseldess) [inline beta-20161006.md]: Done.

**Pull request merged**